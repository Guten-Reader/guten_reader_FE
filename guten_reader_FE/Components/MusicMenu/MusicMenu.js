import React, {Component} from 'react';
import { View, TouchableOpacity, Image, StyleSheet } from 'react-native';
import '../../assets/volume-on.png'
import { createStackNavigator } from 'react-navigation-stack';
import { createAppContainer } from 'react-navigation';

export default class MusicMenu extends Component {
  
  state = {isMuted: false}

  toggleSound = () => {
    this.state.isMuted ? this.setState({isMuted: false}) : this.setState({isMuted: true})
  }

  render() {
    return(
      <View style={styles.container}>
        <TouchableOpacity onPress={this.toggleSound}>
          {this.state.isMuted === true && <Image style={styles.volume} source={require('../../assets/volume-on.png')} />}
        </TouchableOpacity>
        <TouchableOpacity onPress={this.toggleSound}>
          {!this.state.isMuted && <Image style={styles.mute} source={require('../../assets/mute.png')} />}
        </TouchableOpacity>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'black',
  },
  volume: {
    height: 50,
    resizeMode: 'contain'
  },
  mute: {
    height: 50,
    resizeMode: 'contain'
  }

});
