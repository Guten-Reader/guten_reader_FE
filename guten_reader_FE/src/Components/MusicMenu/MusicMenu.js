import React, {Component} from 'react';
import { View, Text, TouchableOpacity, Image, StyleSheet, Button } from 'react-native';
// import '../../../assets/volume-on.png'
import { createStackNavigator } from 'react-navigation-stack';
import { createAppContainer, withNavigation } from 'react-navigation';

class MusicMenu extends Component {

  render() {
    // const track = ^^^ song URL in this.state.currentSong or whatever
    // gets confusing:
    //    const video = this.state.isChanging ? null : (
      // <Video source={{uri: track.audioURL}}
      // paused={this.state.paused} pauses playback
      // maybe move sound/mute to two separate buttons, when one is pushed, isPaused is true, whatever
    // )
    return(
      <View style={styles.toolbar}>
        {/* <Button style={styles.button} onPress={() => this.props.navigation.navigate('Library')} title="BACK"></Button>
        <TouchableOpacity onPress={this.toggleSound}>
          {this.state.isMuted === true && <Image style={styles.volume} source={require('../../assets/volume-on.png')} />}
        </TouchableOpacity>
        <TouchableOpacity onPress={this.toggleSound}>
          {!this.state.isMuted && <Image style={styles.mute} source={require('../../assets/mute.png')} />}
        </TouchableOpacity> */}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  toolbar: {
    flex: 1,
    flexDirection: 'row',
    position: 'absolute',
    bottom: 0,
    right: 0,
    left: 0,
    backgroundColor: '#53E69B',
    justifyContent: 'center',
    height: 65,
    width: '100%',
    alignItems: 'center'
  },
  volume: {
    height: 50,
    width: 50,
    resizeMode: 'contain',
    marginLeft: 100
  },
  mute: {
    height: 50,
    width: 50,
    resizeMode: 'contain',
    marginLeft: 100
  }

});

export default withNavigation(MusicMenu);
