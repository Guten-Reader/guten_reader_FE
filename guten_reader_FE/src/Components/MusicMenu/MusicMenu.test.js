import React from 'react';
import { shallow } from 'enzyme';

import MusicMenu from './MusicMenu';

describe('MusicMenu renders correctly with Jest', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MusicMenu />);
    expect(wrapper).toMatchSnapshot();
  })
})