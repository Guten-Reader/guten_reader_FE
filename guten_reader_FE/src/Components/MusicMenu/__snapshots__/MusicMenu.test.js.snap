// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicMenu renders correctly with Jest renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
